npm/ng-packs/packages/core/src/lib/services/router-events.service.ts

```diff
@@ -19,12 +20,32 @@ export const NavigationEvent = {
 
 @Injectable({ providedIn: 'root' })
 export class RouterEvents {
-  constructor(private router: Router) {}
+  protected readonly router = inject(Router);
 
-  getEvents<T extends RouterEventConstructors>(...eventTypes: T) {
+  readonly #lastNavigation = signal<string | undefined>(undefined);
+  lastNavigation = this.#lastNavigation.asReadonly();
+
+  constructor() {
+    this.listenToNavigation();
+  }
+
+  protected listenToNavigation(): void {
+    this.router.events.pipe(filter(e => e instanceof NavigationEvent.End)).subscribe(() => {
+      // It must be "NavigationTransition" but it is not exported in Angular
+      //https://github.com/angular/angular/blob/9c486c96827a9282cbdbff176761bc95554a260b/packages/router/src/navigation_transition.ts#L282
+      const lastNavigation = this.router.lastSuccessfulNavigation as unknown as any;
+      const curr = lastNavigation.targetRouterState as RouterState;
+      const currUrl = curr.snapshot.url;
```

Comment:
When this observable emits new value, i expected to have lastSuccessfulNavigation as the previous navigation URL. But it doesn't work like that. It returns the current URL which we can access it from the `e` too.

This solution is not working when navigating back.